src/ripple/basics/impl/PerfLogImp.cpp

```diff
@@ -477,14 +477,12 @@ PerfLogImp::onStop()
         }
         thread_.join();
     }
-    if (areChildrenStopped())
-        stopped();
 }
 
 void
 PerfLogImp::onChildrenStopped()
 {
-    onStop();
```

Comment:
These changes should exhibit exactly the same behavior, but I think are easier to read. When `onChildrenStopped()` calls `onStop()`, `thread_.joinable()` should return `false` because it was joined in the earlier call to `onStop` (which is called before `onChildrenStopped` in the shutdown sequence). Thus, the first conditional branch is skipped. The second conditional tests `areChildrenStopped()` which returns the value of `m_childrenStopped`, which starts as `false`, and is only set to `true` once, immediately before calling `onChildrenStopped`, which occurs some time after the call to `onStop`. So in `onStop`, `areChildrenStopped()` should always return `false`, and in `onChildrenStopped`, `areChildrenStopped()` should always return `true`.